modules/apps/27-interchain-accounts/keeper/handshake_test.go

```diff
@@ -290,3 +295,62 @@ func (suite *KeeperTestSuite) TestOnChanOpenConfirm() {
 		})
 	}
 }
+
+func (suite *KeeperTestSuite) TestChanOpenAck() {
+	// no middleware case
+	suite.SetupTest() // reset
+	path := NewICAPath(suite.chainA, suite.chainB)
+	owner := "owner"
+	counterpartyVersion := types.Version
+	suite.coordinator.SetupConnections(path)
+
+	err := InitInterchainAccount(path.EndpointA, owner)
+	suite.Require().NoError(err)
+
+	err = path.EndpointB.ChanOpenTry()
+	suite.Require().NoError(err)
+
+	err = path.EndpointA.ChanOpenAck()
+	suite.Require().NoError(err)
+
+	// middleware case
+
+	suite.SetupTest() // reset
+	path = NewICAPath(suite.chainA, suite.chainB)
+	owner = "owner"
+	suite.coordinator.SetupConnections(path)
+
+	err = InitInterchainAccount(path.EndpointA, owner)
+	suite.Require().NoError(err)
+
+	err = path.EndpointB.ChanOpenTry()
+	suite.Require().NoError(err)
+
+	// add base app
+	mockModule, ok := suite.chainA.GetSimApp().GetModuleManager().Modules["mock"].(mock.AppModule)
+	suite.Require().True(ok)
+	mockModule.Middleware.OnChanOpenAck = func(ctx sdk.Context, portID, channelID, version string) error {
+		if channelID == "channel-0" {
+			return fmt.Errorf("error")
+		}
+		return nil
+	}
```

Comment:
main code needed for developers to add their own custom logic to mock callbacks for testing. First line will be made into `GetMockModule` function in simapp